### DISCUSSION TITLE
Proposal: Allow plugins to use expressions without enabling alerting

### DISCUSSION THREAD

**Discussion by avidhanju** [grafana/grafana#48023] [Plugins]
## Problem
In order to enable expressions, we need to set `alerting: true` in plugin.json file. However, doing this also enables the alerting feature set for the data source.

## Proposal 

Allow plugin authors to enable expressions without enabling alerting.

- Having a separate flag in plugin.json (perhaps `expressions: true/false`) that only enables expressions. 
- Ability to show/hide +Expression button from QueryEditor would be extremely useful. This will help in cases where the query is going to return data that is not going to be compatible with expressions. For example, hierarchical data intended to be visualized with Node Graph visualization.
- Ability to filter the type of Operations that the query supports. For example, only show Math as the supported operation.

**Comment by avidhanju** [comment]
Related bug #48032

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
